sympy/printing/latex.py

```diff
@@ -1415,6 +1417,55 @@ def _print_ZeroMatrix(self, Z):
     def _print_Identity(self, I):
         return r"\mathbb{I}"
 
+    def _print_NDimArray(self, expr):
+
+        mat_str = self._settings['mat_str']
+        if mat_str is None:
+            if self._settings['mode'] == 'inline':
+                mat_str = 'smallmatrix'
+            else:
+                if (expr.rank() == 0) or ((expr.shape[-1] <= 10) == True):
```

Comment:
SymPy inequalities may remain unevaluated, this way avoids that case. That may occur in case of symbolic shape. I'm not sure on whether we will ever enable N-dim arrays to support a symbolic shape, because there's `Indexed` for that.

I copied this code from the _Matrix_ printer and replaced `is` with `==`.